src/main/scala/mesosphere/marathon/api/LeaderProxyFilter.scala

```diff
@@ -186,6 +187,10 @@ class JavaUrlConnectionRequestForwarder @Inject() (
 
   private[this] val viaValue: String = s"1.1 $myHostPort"
 
+  private lazy val ignoreHostnameVerifier = new javax.net.ssl.HostnameVerifier {
+    def verify(hostname: String, sslSession: SSLSession): Boolean = true
```

Comment:
is it worth tracking a set of the last N hostnames and logging a warning whenever we get a hostname that's not already in the set?